addons/mail/static/src/discuss/gif_picker/gif_picker.xml

```diff
@@ -0,0 +1,77 @@
+                <input class="form-control border-0 flex-grow-1" placeholder="Search for a gif" t-on-input="onInput" t-model="state.searchTerm" t-ref="autofocus" />
+                <i class="oi oi-search p-2 fs-5" title="Search..." role="img" aria-label="Search..."></i>
+            </div>
+            <div class="overflow-y-auto overflow-x-hidden" t-ref="scroller">
+                <div t-if="state.loadingError" class="text-center">
+                    <span class="o-discuss-gif-error d-block">😵‍💫</span>
+                    <span class="fs-5 text-muted d-block">Failed to load gifs...</span>
+                </div>
+                <div t-if="state.showCategories and !state.loadingError" class="o-discuss-GifPicker-category-list row row-cols-2 gy-2 gx-2">
+                    <div t-if="!store.guest" class="col">
+                        <div class="o-discuss-GifPicker-category position-relative ratio ratio-16x9" t-on-click="onClickFavoriteCategory">
+                            <img
+                                t-if="state.favorites.length > 0"
+                                class="o-discuss-GifPicker-category"
+                                t-att-src="state.favorites[0].media_formats.tinygif.url"
+                                loading="lazy"
+                            />
+                            <div class="position-absolute w-100 h-100 top-0 start-0 text-center o-mail-text-white align-middle fs-2 bg-black bg-opacity-50">
+                                Favorites
+                            </div>
+                        </div>
+                    </div>
+                    <t t-foreach="state.categories" t-as="category" t-key="category.name">
+                        <div class="col">
+                            <div class="o-discuss-GifPicker-category position-relative ratio ratio-16x9" t-on-click="() => this.onClickCategory(category)">
+                                <img class="o-discuss-GifPickerCategory-img img-fluid" t-att-src="category.image" loading="lazy"/>
+                                <div class="position-absolute w-100 h-100 top-0 start-0 text-center o-mail-text-white align-middle fs-2 bg-black-50">
+                                    <t t-out="category.name"/>
+                                </div>
+                            </div>
+                        </div>
+                    </t>
+                </div>
+                <t t-if="state.gifs and !state.showCategories">
+                    <div class="d-flex gap-1">
+                        <div>
+                            <t t-foreach="evenGif" t-as="gif" t-key="gif_index">
+                                <t t-call="discuss.Gif"/>
+                            </t>
+                        </div>
+                        <div>
+                            <t t-foreach="oddGif" t-as="gif" t-key="gif_index">
+                                <t t-call="discuss.Gif"/>
+                            </t>
+                        </div>
```

Comment:
The strategy to have 2 columns and one holds even and the other odd gifs is good when a few gifs, but when you do many load more, columns is heavily unbalanced and only half of them are shown in a single column:
https://github.com/odoo/odoo/assets/6569390/6fa1336c-f573-4151-8571-d3e368a217c1

I would suggest the following strategy: whenever you load more gifs, you push each of them in either column 1 or column 2 depending on whichever is the smallest height before push. The size of column is the sum of all gifs being pushed to this column. You can keep just 2 numbers and add height of last inserted gif in the appropriate column.

If you do this, then the columns are always somewhat balanced, and it works fine.

In a previous comment I said that load more at the very bottom works but it's not great, as we feel that we should do many scroll to bottom attempts before it actually loads the gifs. If you more the threshold to trigger load more higher, then it will load more at a point where both columns looked filled, so columns will look fully balanced from end-user perspective.

Reply:
I implemented this, it works nicely IMHO :)